Navigator edits

PhillMyCup opened the pull request.

MrFobwatch committed:
Navigator Outline

Updated the Navigator Class with descriptions of intended functionality.

MrFobwatch committed:
Merged branch master into navigator-edits

MrFobwatch committed:
Created Navigator Source

Implemented the pseudo code from Slack. Still not in working state.

MrFobwatch committed:
Created Navigator Source

Implemented the pseudo code from Slack. Still not in working state.

MrFobwatch committed:
Merged branch navigator-edits into navigator-edits

PhillMyCup committed:
Changes to gps and navigator

GPS - Added askForDirections
Navigator - updated find unvisited cell

MrFobwatch committed:
Updated Navigator Classes to new code

implemented changes from Maze class

MrFobwatch committed:
Added Some Documenation for the Navigator

Navigator adjustments and some GPS changes

PhillMyCup committed:
Updates to Navigator

MrFobwatch committed:
Change the way center is checked by navigator

MrFobwatch committed:
Outline more commands for Navigator

PhillMyCup committed:
Fixed up Navigator.cpp

deleted findbox();  
fixed up findUnvisitedCell();
created map();

PhillMyCup committed:
Update Navigator

tbhova committed:
Added documentation

PhillMyCup committed:
Update Navigator & Driver

MrFobwatch committed:
Create Intial Navigator Test

MrFobwatch committed:
Create Intial Navigator Test

MrFobwatch committed:
Merged branch navigator-edits into navigator-edits

tbhova commented on tests/auto/micromouse/navigator_test.h.autosave:
```diff
@@ -0,0 +1,22 @@
+#ifndef NAVIGATOR_TEST_H
```
Delete this entire file.
Kevin probably has the autosave version still that needs saved

MrFobwatch replied:
Whoops. I'll do that shortly.

tbhova commented on tests/auto/micromouse/navigator_test.h:
```diff
@@ -0,0 +1,22 @@
+class NavigatorTest : public ::testing::Test {
+    
+};
+
```
Add at least one test, something like:
```
TEST_F(NavigatorTest, testMapping) {

}
```

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.h:
```diff
@@ -11,18 +11,40 @@
+    Driver driver;
+    GPS directions;
+    Maze maze;
+    Coordinate destination;
```
I don't think this needs to be a member variable. Just make a new variable inside a function any time your need it.

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.h:
```diff
@@ -11,18 +11,40 @@
+    GPS directions;
+    Maze maze;
+    Coordinate destination;
+    void updateMaze(Coordinate cell, std::vector<Cardinal8> walls); //Get wall data from Driver and adjust maze
```
Change this to Doxygen Style documentation

MrFobwatch committed:
Add a Navigator Test Prototype

tbhova committed:
Removed documentation and moved it to the documentation repo

tbhova commented on src/src/navigator.cpp:
```diff
@@ -0,0 +1,51 @@
+#include "navigator.h"
+using namespace std;
+Coordinate Navigator::findUnvisitedCell() {
+    for (int x = 0; x < sizeX; x++) {
```
sizeX should be maze.getSizeX()

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.cpp:
```diff
@@ -0,0 +1,51 @@
+using namespace std;
+Coordinate Navigator::findUnvisitedCell() {
+    for (int x = 0; x < sizeX; x++) {
+        for (int y = 0; y < sizeY; y++){
```
same thing for sizeY

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.cpp:
```diff
@@ -0,0 +1,51 @@
+        for (int y = 0; y < sizeY; y++){
+            if (!maze.hasMouseVisited(x,y)){
+                return Coordinate;
+                break;
```
No break needed since we return

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.cpp:
```diff
@@ -0,0 +1,51 @@
+
+
+bool Navigator::isDestination(int x, int y) {
+    return isCenter(x) && isCenter(y);
```
isCenter is not defined anywhere in this cpp file

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.h:
```diff
@@ -11,18 +11,40 @@
+     * @return Coordinates of the the nearest UnvisitedCell
+     */
+    Coordinate findUnvisitedCell();
+        //Search maze for a cell.visited = false
```
These comments can either be deleted or added to the method's @brief

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.h:
```diff
@@ -11,18 +11,40 @@
+        //determine coordinates of cell
+
+    bool isDestination(Coordinate cell); //Returns whether a cell is the destination
+    bool isCenter(int i);
```
This method needs defined in the cpp

MrFobwatch replied:
Resolved

tbhova commented on src/src/navigator.h:
```diff
@@ -11,18 +11,40 @@
+
+    //this will be called by main - findUnvisitedCell with maze,
+    //use GPS to find direction to that cell, inform driver of direction
+    //Loop this until all cells are visited
```
Change to Doxygen style

MrFobwatch replied:
Resolved

tbhova reviewed (changes requested):
Address these comments and submit a new pull request

tbhova committed:
Fixed parameter naming

tbhova committed:
Updated Navigator's maze, driver, and gps

MrFobwatch committed:
Address review commments

tbhova committed:
Merge remote-tracking branch 'refs/remotes/origin/navigator-edits'

Conflicts:
	src/src/navigator.h

tbhova committed:
Made navigator build... almost

tbhova committed:
Added constructor so project builds

MrFobwatch committed:
Create Maze Map Test

MrFobwatch committed:
Resolves #12

Checks if the maze is mapped and if it is the destination is set to origin.

tbhova committed:
Almost builds

tbhova committed:
Merge remote-tracking branch 'origin/navigator-edits' into navigator-edits

tbhova committed:
Total number of cells is only calculated once now.

tbhova committed:
Fixed a bug

tbhova committed:
Navigator builds and the tests run now

tbhova committed:
Removed abstract drive implementation file and fixed some compiler bugs

tbhova committed:
Navigator now actually gets a maze, gps, and driver

tbhova committed:
Clean up + created mock gps in tests

tbhova committed:
Made map loop properly, implement optimal router and run in Navigator

tbhova committed:
Fixed some bugs, added mocked navigator tests

Maze had a static size bug, navigator didn't set cells visited, and navigator didn't loop during map.
